pkg/service/contract/contract.go

```diff
@@ -0,0 +1,681 @@
+						if yaml.Contains(duplicateServices, service) {
+							isAppend = true
+						} else {
+							duplicateServices = append(duplicateServices, service)
+						}
+
+						// Convert the HTTP mock to OpenAPI documentation.
+						openapi, err := s.HTTPDocToOpenAPI(s.logger, *mock)
+						if err != nil {
+							utils.LogError(s.logger, err, "failed to convert the yaml file to openapi")
+							return fmt.Errorf("failed to convert the yaml file to openapi")
+						}
+
+						// Validate the generated OpenAPI schema.
+						err = validateSchema(openapi)
+						if err != nil {
+							return err
+						}
+
+						// Write the OpenAPI document to the specified directory.
+						err = s.openAPIDB.Write(ctx, s.logger, filepath.Join(keployFolder, "schema", "mocks", service, testSetID), "mocks", openapi, isAppend)
+						if err != nil {
+							return err
+						}
+
+						mappingFound = true // Mark the mapping as found.
+						break
+					}
+				}
+
+				// Break the outer loop if the relevant mapping is found.
+				if mappingFound {
+					break
+				}
+			}
+		}
+	}
+
+	return nil // Return nil if the function completes successfully.
+}
+
+func (s *contract) GenerateTestsSchemas(ctx context.Context, selectedTests []string) error {
+	keployFolder := "./keploy/"
+	testSetsIDs, err := s.testDB.GetAllTestSetIDs(ctx)
+	if err != nil {
+		utils.LogError(s.logger, err, "failed to get test set IDs")
+		return err
+	}
+
+	for _, entry := range testSetsIDs {
+		testSetID := entry
+		if !yaml.Contains(selectedTests, testSetID) && len(selectedTests) != 0 {
+			continue
+		}
+
+		testCases, err := s.testDB.GetTestCases(ctx, testSetID)
+		if err != nil {
+			utils.LogError(s.logger, err, "failed to get test cases", zap.String("testSetID", testSetID))
+			return err
+		}
+		for _, tc := range testCases {
+			var httpSpec models.HTTPDoc
+			httpSpec.Kind = string(tc.Kind)
+			httpSpec.Name = tc.Name
+			httpSpec.Spec.Request = tc.HTTPReq
+			httpSpec.Spec.Response = tc.HTTPResp
+			httpSpec.Version = string(tc.Version)
+
+			openapi, err := s.HTTPDocToOpenAPI(s.logger, httpSpec)
+			if err != nil {
+				s.logger.Error("Failed to convert the yaml file to openapi")
+				return fmt.Errorf("failed to convert the yaml file to openapi")
+			}
+			// Validate the OpenAPI document
+			err = validateSchema(openapi)
+			if err != nil {
+				return err
+			}
+			// Save it using the OpenAPIDB
+			err = s.openAPIDB.Write(ctx, s.logger, filepath.Join(keployFolder, "schema", "tests", entry), tc.Name, openapi, false)
+			if err != nil {
+				return err
+			}
+
+		}
+
+	}
+	return nil
+}
+
+func (s *contract) Generate(ctx context.Context) error {
+	if checkConfigFile(s.config.Contract.ServicesMapping) != nil {
+		utils.LogError(s.logger, fmt.Errorf("Error in checking config file while validating"), "Error in checking config file while validating")
+		return fmt.Errorf("Error in checking config file while validating")
+	}
+
+	var config config.Config
```

Comment:
Why are you still reading this????? When you already have this config file in the struct.